src/provisioning/Provisioner.ts

```diff
@@ -877,40 +884,42 @@ export class Provisioner {
             throw new Error("Server requested for linking not found");
         }
 
-        // Make sure the requester is a mod in the room
-        const botCli = this.ircBridge.getAppServiceBridge().getBot().getClient();
-        const stateEvents = await botCli.roomState(roomId);
-        // user_id must be JOINED and must have permission to modify power levels
-        let isJoined = false;
-        let hasPower = false;
-        stateEvents.forEach((e: { type: string; state_key: string; content: {
-            state_default?: number;
-            users_default?: number;
-            membership: string;
-            users?: Record<string, number>;
-            events?: Record<string, number>;
-        };}) => {
-            if (e.type === "m.room.member" && e.state_key === options.user_id) {
-                isJoined = e.content.membership === "join";
-            }
-            else if (e.type === "m.room.power_levels" && e.state_key === "") {
-                // https://matrix.org/docs/spec/client_server/r0.6.0#m-room-power-levels
-                let powerRequired = e.content.state_default || 50; // Can be empty. Assume 50 as per spec.
-                if (e.content.events && e.content.events["m.room.power_levels"]) {
-                    powerRequired = e.content.events["m.room.power_levels"];
+        if (!ignorePermissions) {
+            // Make sure the requester is a mod in the room
+            const botCli = this.ircBridge.getAppServiceBridge().getBot().getClient();
+            const stateEvents = await botCli.roomState(roomId);
```

Comment:
Not for now, but this is pants and horribly bad for performance. We should really just fetch the two state events we care about. Up to you if you fancy fixing this here or if we should make an issue.

Reply:
I also wonder if this should be part of Intent.

Reply:
What two events?
If we were to fetch a state event where the person became a moderator, wouldn't we also need to check all ways they may have lost that permission?